lua/telescope/_extensions/file_browser/picker.lua

```diff
@@ -25,6 +25,31 @@ local os_sep = Path.path.sep
 -- enclose in module for docgen
 local fb_picker = {}
 
+-- try to get the index of entry of current buffer
+local get_selection_index = function(opts, results)
+  local init_dir = Path:new(opts.path):absolute()
+  local buf_path = Path:new(vim.api.nvim_buf_get_name(0)):absolute()
+  local current_dir = Path:new(buf_path):parent():absolute()
+
+  if init_dir ~= current_dir then
+    return 1
+  end
+
+  for i, v in ipairs(results) do
+    local p = Path:new(v.value):absolute()
```

Comment:
value should correspond to absolute paths; hence `Path:new` is not required here.